presto-main/src/main/java/com/facebook/presto/execution/TaskExecutionId.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import com.facebook.drift.annotations.ThriftConstructor;
+import com.facebook.drift.annotations.ThriftField;
+import com.facebook.drift.annotations.ThriftStruct;
+import com.facebook.presto.spi.QueryId;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.facebook.presto.spi.QueryId.parseDottedId;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Integer.parseInt;
+import static java.util.Objects.requireNonNull;
+
+@ThriftStruct
+public class TaskExecutionId
+{
+    private final TaskId taskId;
```

Comment:
Was following the same pattern as Stage , StageExecutionId. But I thought about it a bit more and will be adding an extra field `attemptNumber` instead of creating a new class. 

Closing this PR out as i have been working on a different branch - https://github.com/prestodb/presto/pull/19649